event-handler/teleport_events_watcher_test.go

```diff
@@ -136,6 +137,129 @@ func TestNext(t *testing.T) {
 	}
 }
 
+// errMockTeleportEventWatcher is Teleport client mock that returns an error after the first SearchUnstructuredEvents
+type errMockTeleportEventWatcher struct {
+	// events is the mock list of events
+	events []events.AuditEvent
+	t      *testing.T
+	called bool
+}
+
+func (c *errMockTeleportEventWatcher) SearchEvents(ctx context.Context, fromUTC, toUTC time.Time, namespace string, eventTypes []string, limit int, order types.EventOrder, startKey string) ([]events.AuditEvent, string, error) {
+	e := c.events
+	c.events = nil
+	return e, "test", nil
+}
+
+func (c *errMockTeleportEventWatcher) StreamSessionEvents(ctx context.Context, sessionID string, startIndex int64) (chan events.AuditEvent, chan error) {
+	return nil, nil
+}
+
+func (c *errMockTeleportEventWatcher) SearchUnstructuredEvents(ctx context.Context, fromUTC, toUTC time.Time, namespace string, eventTypes []string, limit int, order types.EventOrder, startKey string) ([]*auditlogpb.EventUnstructured, string, error) {
+	if c.called {
+		return nil, "", errors.New("error")
+	}
+
+	e := c.events
+	c.events = nil
+
+	events := make([]*auditlogpb.EventUnstructured, len(e))
+	for i, event := range e {
+		events[i] = eventToJSON(c.t, event)
+	}
+	c.called = true
+	return events, "", nil
+}
+
+func (c *errMockTeleportEventWatcher) StreamUnstructuredSessionEvents(ctx context.Context, sessionID string, startIndex int64) (chan *auditlogpb.EventUnstructured, chan error) {
+	return nil, nil
+}
+
+func (c *errMockTeleportEventWatcher) UpsertLock(ctx context.Context, lock types.Lock) error {
+	return nil
+}
+
+func (c *errMockTeleportEventWatcher) Ping(ctx context.Context) (proto.PingResponse, error) {
+	return proto.PingResponse{
+		ServerVersion: Version,
+	}, nil
+}
+
+// Close is mock close method
+func (c *errMockTeleportEventWatcher) Close() error {
+	return nil
+}
+
+func newErrTeleportEventWatcher(t *testing.T, e []events.AuditEvent, exitOnLastEvent bool) *TeleportEventsWatcher {
+	teleportEventWatcher := &errMockTeleportEventWatcher{events: e, t: t}
+
+	client := &TeleportEventsWatcher{
+		client: teleportEventWatcher,
+		pos:    -1,
+		config: &StartCmdConfig{
+			IngestConfig: IngestConfig{
+				BatchSize:       5,
+				ExitOnLastEvent: exitOnLastEvent,
+			},
+		},
+	}
+
+	return client
+}
+
+func TestLastEvent(t *testing.T) {
+	t.Run("should not leave hanging go-routines", func(t *testing.T) {
+		const (
+			nIters = 5
+		)
```

Comment:
I would move this right before the loop where it is first used, and declare it on a single line.

```
const nIters = 5
```